ee/tabby-db/src/email_service_credential.rs

```diff
@@ -44,21 +44,15 @@ impl DbConn {
         smtp_password: Option<String>,
         smtp_server: String,
     ) -> Result<()> {
-        let smtp_password = match smtp_password {
-            Some(pass) => pass,
-            None => {
-                self.read_email_service_credential()
-                    .await?
-                    .ok_or_else(|| {
-                        anyhow!("No existing SMTP credentials, password must be specified")
-                    })?
-                    .smtp_password
-            }
-        };
-
         Ok(self
             .conn
             .call(move |c| {
+                 let smtp_password = match smtp_password {
```

Comment:
I don't think this ensures the transaction semantic.

See https://docs.rs/rusqlite/latest/rusqlite/struct.Transaction.html for an example